src/syrupy/serializers/amber.py

```diff
@@ -146,7 +163,11 @@ def serialize_dict(
 
     @classmethod
     def serialize_iterable(
-        cls, data: "SerializableData", *, depth: int = 0, visited: Set[Any] = set()
+        cls,
+        data: "SerializableData",
+        *,
+        depth: int = 0,
+        visited: Optional[Set[Any]] = None,
```

Comment:
Why the refactor?

Reply:
Previously we set the default arg to set(). This does not work like you'd expect. It doesn't create a "set" every function invocation, but rather re-uses a single set